js/product-details.js

```diff
@@ -0,0 +1,155 @@
+var xmlhttp = new XMLHttpRequest();
+xmlhttp.onreadystatechange = function () {
+  if (this.readyState == 4 && this.status == 200) {
+    var data = JSON.parse(this.responseText);
+    var drink = data[productCategory][productId].additionalDescription;
+    for (key in drink) {
+      if (drink[key] === "") {
+      } else {
+        buildDetailButton(drink);
+        bildDetailvaluecontainers(drink);
+      }
+    }
+    clickButton(drink);
+    const PRODUCT_DETAILS_CHANGES_POSITION_WHEN_WIDTH = 1000;
+    if (windowWidth() > PRODUCT_DETAILS_CHANGES_POSITION_WHEN_WIDTH) {
+      let activeFirstButton = document.getElementsByClassName("details-description")[0];
+      activeFirstButton.click();
+    }
+  }
+};
+xmlhttp.open("GET", "../jsons/" + productCategory + ".json", true);
+xmlhttp.send();
+
+const buildDetailButton = function (drink) {
+  var productDetails = document.getElementsByClassName("product-details")[0];
+  let detailButton = document.createElement("div");
+  detailButton.innerHTML = key;
+  detailButton.classList.add("details-" + key);
+  detailButton.classList.add("details-button");
+  productDetails.appendChild(detailButton);
+};
+
+const bildDetailvaluecontainers = function (drink) {
+  var containerColumn = document.getElementsByClassName("product-details")[0];
+  var containerRow = document.getElementsByClassName(
+    "product-details-value"
+  )[0];
+  if (typeof drink[key] == "object" && Array.isArray(drink[key]) != true) {
+    containerIfValueObject(drink[key], containerColumn, containerRow)
+  } else if (Array.isArray(drink[key])) {
+    containerIfValueArray(drink[key], containerColumn, containerRow)
+  } else {
+    containerIfValueLine(drink[key], containerColumn, containerRow);
+  }
+};
+
+const containerIfValueObject = function (value, containerColumn, containerRow) {
+  let drinkKey = value;
+  let valuecontainerColumn = document.createElement("div");
+  let valuecontainerRow = document.createElement("div");
+  for (key2 in drinkKey) {
+    if (drinkKey.hasOwnProperty(key2)) {
```

Comment:
BTW (by the way) it's important thing - you should check hasOwnProperty when you do for in or for of
there're a few places with for in more in your code - add this condition to them too